lib/libzfs/libzfs_dataset.c

```diff
@@ -4028,6 +4028,24 @@ zfs_destroy_snaps_nvl(libzfs_handle_t *hdl, nvlist_t *snaps, boolean_t defer)
 			    dgettext(TEXT_DOMAIN, "snapshot is cloned"));
 			ret = zfs_error(hdl, EZFS_EXISTS, errbuf);
 			break;
+		case EBUSY: {
+			nvlist_t *existing_holds;
+			lzc_get_holds(nvpair_name(pair), &existing_holds);
```

Comment:
Looking again, all other callers of `lzc_get_holds()` check for possible errors.  Without it you'll likely get some uninitialized memory dereference in `nvlist_empty()` below.